torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1571,6 +1585,12 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         # Start of a forward pass.
         self.training_state = TrainingState_.FORWARD
 
+        if self._is_root:
+            if self.device_id:
```

Comment:
Should we change to this `if self.device_id is not None`? It might confound `self.device_id == 0` and `self.device_id is None` right now.